test/dotnet.Tests/CommandTests/Test/TestProgressStateTests.cs

```diff
@@ -0,0 +1,195 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.DotNet.Cli.Commands.Test;
+using Microsoft.DotNet.Cli.Commands.Test.Terminal;
+using Moq;
+using Xunit;
+
+namespace dotnet.Tests.CommandTests.Test;
+
+public class TestProgressStateTests
```

Comment:
@nohwnd @mariam-abdulla Any ideas for additional test cases?

Reply:
yup, the ones I provided :))